newlib/libc/sys/vita/fcntl.c

```diff
@@ -114,7 +117,34 @@ int _fcntl_r(struct _reent *reent, int fd, int cmd, ...)
 		}
 	}
 
+	if (fdmap->type == VITA_DESCRIPTOR_FILE)
+	{
+		if (cmd == F_GETFL)
+		{
+			int flags = fdmap->flags;
+			__vita_fd_drop(fdmap);
+			return flags;
+		}
+
+		if (cmd == F_SETFL)
+		{
+			// File access modes should be ignored
+			arg = arg & (O_NONBLOCK | O_APPEND | O_DIRECT | O_ASYNC | O_SYNC | O_DSYNC);
```

Comment:
more great for me now.

one more thing.

arg should be
```
SUPPORT = (O_NONBLOCK | O_APPEND | O_DIRECT | O_ASYNC | O_SYNC | O_DSYNC);

arg = (arg & SUPPORT) | (fdmap->flags & ~SUPPORT);
```
i'm not sure, it's correct or not.
but `ignore` means not happend anything instead removing.

Reply:
Yeah, you're right, we should keep initial file access flags as is
Pushed update for that